### PR TITLE
Issue 99 - Recycler view item visibility

### PR DESCRIPTION
**Pull request by heitorcolangelo**
[#99](https://github.com/concretesolutions/kappuccino/issues/99)

This is a workaround for this issue. Unfortunately I could not find myself how to properly fixed it, but I provided a workaround and proper advices and documentation while this is not fixed. This should solve the specific case for now.
Also, I've updated some library versions and organised the code a lil bit.

### CODE REVIEW

**Review by matheus-corregiari — APPROVED** [review]

File: kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt [hunk 1]
```diff
@@ -68,16 +72,79 @@ class RecyclerViewMethods(private val recyclerViewId: Int) {
...
+         * Anyways, use carefully or, use one the following methods instead:
+         *
+         * @use [displayed] to assert that the whole item is displayed.
+         * @use [displayedChildView] to assert that a specific child view of that item is displayed.
+         *
+         * @deprecated use [displayed] or [displayedChildView] instead.
+         */
+        @Deprecated(
+                level = DeprecationLevel.WARNING,
+                message = "User displayed() or displayedChildView(childViewId) instead")
```
Comment (heitorcolangelo) [inline kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt]: typo **User**, should be **Use**

File: kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt [hunk 2]
```diff
@@ -68,16 +72,79 @@ class RecyclerViewMethods(private val recyclerViewId: Int) {
             return this
         }
 
+        /**
+         * This method is deprecated until [this issue](https://github.com/concretesolutions/kappuccino/issues/99) is solved.
+         * I do not recommend the use when you try to match using the Id of the view. But it seems to work fine if
+         * you use the Text instead of the Id.
+         * Anyways, use carefully or, use one the following methods instead:
```
Comment (heitorcolangelo) [inline kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt]: "...use one **of** the following..."

File: kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt [hunk 3]
```diff
@@ -68,16 +72,79 @@ class RecyclerViewMethods(private val recyclerViewId: Int) {
...
+        fun displayedChildView(@IdRes childViewId: Int): Interactions {
+            onView(withId(recyclerViewId)).displayedChildView(position, childViewId)
+            return this
+        }
+
+        /**
+         * This method is deprecated until [this issue](https://github.com/concretesolutions/kappuccino/issues/99) is solved.
+         * I do not recommend the use when you try to match using the Id of the view. But it seems to work fine if
+         * you use the Text instead of the Id.
+         * Anyways, use carefully or, use one the following methods instead:
```
Comment (heitorcolangelo) [inline kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt]: "...use one **of** the following..."

File: kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt [hunk 4]
```diff
@@ -68,16 +72,79 @@ class RecyclerViewMethods(private val recyclerViewId: Int) {
...
+        fun notDisplayedChildView(@IdRes childViewId: Int): Interactions {
+            onView(withId(recyclerViewId)).notDisplayedChildView(position, childViewId)
+            return this
+        }
+
+        /**
+         * This method is deprecated until [this issue](https://github.com/concretesolutions/kappuccino/issues/99) is solved.
+         * I do not recommend the use when you try to match using the Id of the view. But it seems to work fine if
+         * you use the Text instead of the Id.
+         * Anyways, use carefully or, use one the following methods instead:
```
Comment (heitorcolangelo) [inline kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/recyclerView/RecyclerViewMethods.kt]: "...use **one** of the following..."

**Review by heitorcolangelo — COMMENTED** [review]
Typos =(

**Pull request merged**